### PR TITLE
Use a better syntax for dispatching generics on R6 Node class

### PR DESCRIPTION
**Pull request by johnkerl**
Implementing great feedback on https://github.com/TileDB-Inc/TileDB-Cloud-R/pull/71#discussion_r797221411

### CODE REVIEW

File: R/manual_layer_node.R
Comment (johnkerl) [inline R/manual_layer_node.R:361]: Reason for change from `delayed_function` to `node`:

* In this package, `Node` and `DAG` are implementation details; the [user-facing surface](https://tiledb-inc.github.io/TileDB-Cloud-R/articles/TaskGraphs.html) centers on factory methods with `delayed` in the name. So, I thought to call this `delayed_function`.
* However, it's not really a function, it's a `Node` object and perhaps calling it `delayed_function` increased, rather than reduced, clarity ...

Comment (eddelbuettel) [reply R/manual_layer_node.R:361]: Maybe `delayed_compute` to express both components?

File: R/manual_layer_node.R
Comment (eddelbuettel) [inline R/manual_layer_node.R:338]: More consistent, I like it.

File: R/manual_layer_node.R
Comment (eddelbuettel) [inline R/manual_layer_node.R:366]: Ahh. Much clearer. Nice.

File: R/manual_layer_node.R
Comment (eddelbuettel) [inline R/manual_layer_node.R:382]: Nice.

File: R/manual_layer_node.R
Comment (eddelbuettel) [reply R/manual_layer_node.R:361]: Or maybe `delayed_*` is only for args?  Your call....

**Review by eddelbuettel — COMMENTED** [review]
This looks good.   Is this PR final and closed?  Or might there be further commits?

**Comment by eddelbuettel** [comment]
@johnkerl I don't mind a post approval `pkgdown` commit (though one could also stick it into a clean separate docs-only follow-up).   I am not sure how much I really like your approach of treating suggested chunks of works as a "river" that flows and receives multiple updates.  To me, one does a chunk of work, draws a line and submits at that line.  And leaves it alone to the extent possible.  To _routinely_ and even _on purpose_ commit after approval is a bit weird.  But whatever works....

**Review by eddelbuettel — APPROVED** [review]

**Comment by johnkerl** [comment]
@eddelbuettel thanks for the feedback! If it's easier I can submit docs-only PRs. Which will increase the number of PRs to review, of course -- whatever's best for you.

**Comment by eddelbuettel** [comment]
We live and evolve and adapt.  Code only and docs only may be easier to access from the top.   As for number of PRs it's tough with fast changing code.   Not too large yet not 'too frequent'.   

But exciting to see the code come along.  Really nice!

**Pull request merged**
